hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java

```diff
@@ -97,7 +98,9 @@ public List<HoodieRecord<T>> deduplicateRecords(
       final T data1 = rec1.getData();
       final T data2 = rec2.getData();
 
-      @SuppressWarnings("unchecked") final T reducedData = (T) data2.preCombine(data1);
+      Properties properties = new Properties();
+      properties.put("schema", schemaString);
+      @SuppressWarnings("unchecked") final T reducedData = (T) data2.preCombine(data1, properties);
```

Comment:
Should we put the schema string into the properties ? `HoodiePayloadConfig` does not include the schema string, how about we change the method signature into `HoodieRecordPayload#preCombine(T oldValue, Schema schema, Properties properties)`, just like what we do to `HoodieRecordPayload#combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema, Properties properties)`

Reply:
In the previous comment, it suggests achieving the same result by only implementing another payload class without changing the top level API in HoodieRecordPayload, so I reuse HoodieRecordPayload#preCombine(T oldValue, Properties properties), WDYT?  if you think it worth implementing a new Interface, I can also fallback into the old way

Reply:
@xushiyan Do you have any comments?

Reply:
@danny0405 @fengjian428 while it does look more aligned with `preCombine` having a new API containing `Schema`, but the timing of this change does not look great. We have a major refactoring close to land in https://github.com/apache/hudi/tree/release-feature-rfc46 with a new public unified API for merge and deprecating HoodieRecordPayload, which makes introducing `preCombine(T oldValue, Schema schema, Properties properties)` very odd. We have to evaluate the business requirement thoroughly to make the decision: option A) keep the old APIs as is for partial update and implement new partial update merger once RFC46 landed; option B) adding a new APIs for preCombine and keep partial update working and supported via the deprecated APIs, while in parallel support partial update in the new merger APIs

i think option B is acceptable in case of strong business requirements. please chime in here @alexeykudinkin @prasannarajaperumal

Reply:
+1 for option B, let's not bind partial update feature with RFC46.

Reply:
Chatted with @prasannarajaperumal and @alexeykudinkin we are good to go with option B